### ISSUE TITLE
MdxRemote.Components cannot have props?

### ISSUE THREAD

**Issue by MatthewCaseres** [hashicorp/next-mdx-remote#94]
Unsure if I am doing something wrong or if the typing is wrong?
<img width="847" alt="Screen Shot 2021-01-07 at 1 46 12 AM" src="https://user-images.githubusercontent.com/43053796/103865805-5fecde80-508a-11eb-9bd0-883a4a2711d9.png">

**Comment by jescalan** [comment]
`MdxRemote.Components` will accept any valid React component. This error indicates that you are not passing in a valid react component. What do your components look like?

**Comment by MatthewCaseres** [comment]
I embed React components as code blocks that contain YAML having the props.

So this code block
```edtech-mcq 
prompt: This is the prompt
solution: This is the solution
answers:
  - hello world
  - hello 2
correct_idx: 0
id: 63bde698-7dc5-4f25-9703-b01b8041e029
```
is processed by a remark plugin
```js
import visit from 'unist-util-visit'
import yaml from 'js-yaml'
const withMCQ = () => (tree) => {
  visit(tree, "code", (node) => {
    if (node.lang === "edtech-mcq") {
      node.type = "jsx";
      const props = yaml.safeLoad(node.value);
      node.value = `<MCQ {...${JSON.stringify(props)}}/>`;
      console.log(node.value)
    }
  });
};
export default withMCQ
```

**Comment by jescalan** [comment]
Ah ok - this is a remark plugin, and not a react component, so it will not work. The `components` argument is intended for react components. Looks like typescript did its job right here!

**Comment by MatthewCaseres** [comment]
Oh ok sorry, I am passing in a react component, I didn't communicate well. This is what the SmartHeading component looks like:
```js
export default function SmartHeading({ id, slug, contents }: { id: string, slug: string, contents: string }) {
  return (
    <h2
      id={slug}
      onClick={() => {console.log(id)}}
    >
      {contents}
    </h2>
  )
}
```
It is a valid react component, but as TS says, not assignable to FunctionComponent<{}> which I think is saying that the component has no props?

**Comment by jescalan** [comment]
I'm not sure - the `FunctionComponent` type is straight from react, so this is outside the scope of this library. If you have a react function component and its not able to match their type I'm not sure how to further debug. I can tell you that we have tests in this library to verify that react components can be passed in without typescript errors though, and they are working, so there must be something unusual going on with your setup?

**Comment by MatthewCaseres** [comment]
![Screen Shot 2021-01-07 at 11 44 45 AM](https://user-images.githubusercontent.com/43053796/103925954-e2eb5480-50dd-11eb-8521-bbc914592ae1.png)

The component is valid React, not assignable to FunctionComponent<{}>. I think the generic passed into FunctionComponent shouldn't be <{}>?

Right now I am okay just not typing the component and provider. Everything I am passing in renders properly, so it is just the typing that causes me trouble.

I can provide a reproduction if needed, or try to make a PR if the attached image indicates a need for an update to the typing.

**Comment by jescalan** [comment]
[The typing](https://github.com/hashicorp/next-mdx-remote/blob/master/types.d.ts#L6) just uses the bare `FunctionComponent` type from react. This should be sufficient to cover any function component, regardless of whether it has props or not, as far as I know.

If you want to trim this down to a minimal reproduction I'd be happy to take a look though!

**Comment by MatthewCaseres** [comment]
![Screen Shot 2021-01-07 at 12 02 34 PM](https://user-images.githubusercontent.com/43053796/103927621-3f4f7380-50e0-11eb-82e5-bf91986112df.png)

It isn't sufficient to use the bare type, you need to include the props as a generic. The props come from the generic you pass in, here is the typing in React:
![Screen Shot 2021-01-07 at 12 03 42 PM](https://user-images.githubusercontent.com/43053796/103927735-745bc600-50e0-11eb-9a7f-70c8433b6b38.png)

**Comment by jescalan** [comment]
I tried adding props to a function in the test suite and was not tagged with an error by vscode. That being said I am not running this directly through the typescript compiler so it might be off...

**Comment by MatthewCaseres** [comment]
Made a code sandbox - 
https://codesandbox.io/s/blazing-cookies-w2voz?file=/pages/index.tsx

**Comment by jescalan** [comment]
Super helpful, I think I have a fix here with special thanks to @jmfury for the suggested type 🙌🏼

**Comment by jescalan** [comment]
Closed by #95
